MaraBot/Core/PresetValidation.cs

```diff
@@ -92,15 +93,46 @@ public static List<string> ValidateOptions(in Preset preset, IReadOnlyDictionary
                     errors.Add($"{kValidationInfoPrefix} '{pair.Key}' belongs to the {Option.ModeToPrettyString(pairMode)} mode, but the selected mode is {Option.ModeToPrettyString(mode)}.");
 
-                // Skip if value doesn't exist
-                foreach (var o in allOptions[pair.Key].Type == OptionType.List
-                        ? Option.ParseList(pair.Value) : new List<string> { pair.Value })
-                    if (!allOptions[pair.Key].Values.ContainsKey(o))
+                // General enum validation
+                if (option is EnumOption)
+                {
+                    var enumOption = option as EnumOption;
+
+                    // Skip if value doesn't exist
+                    foreach (var o in option.List ? Option.ParseList(pair.Value) : new List<string> { pair.Value })
+                        if (!enumOption.Values.ContainsKey(o))
+                        {
+                            errors.Add($"{kValidationErrorPrefix} '{pair.Key}' has no known value '{o}'");
+                            optionsCopy.Remove(pair.Key); // Make sure we don't validate on unknown values
+                            continue; // Check if there are more unknown values, so that we can show those errors too
+                        }
+                }
+
+                // General numeric validation
+                if (option is NumericOption)
+                {
+                    var numericOption = option as NumericOption;
+
+                    // Skip if value is not within bounds
+                    foreach (var o in option.List ? Option.ParseList(pair.Value) : new List<string> { pair.Value })
                     {
-                        errors.Add($"{kValidationErrorPrefix} '{pair.Key}' has no known value '{o}'");
-                        optionsCopy.Remove(pair.Key); // Make sure we don't validate on unknown values
-                        continue;
+                        double v;
+                        bool success = double.TryParse(o, out v);
+                        if(!success)
+                        {
+                            errors.Add($"{kValidationErrorPrefix} '{pair.Key}' has non-numeric value '{o}'");
+                            optionsCopy.Remove(pair.Key); // Make sure we don't validate on invalid values
+                            continue; // Check if there are more invalid values, so that we can show those errors too
+                        }
+                        if (v < numericOption.Min || v > numericOption.Max)
+                        {
+                            errors.Add($"{kValidationErrorPrefix} '{pair.Key}' has out-of-bounds value '{o}' (must be between {numericOption.Min} and {numericOption.Max})");
+                            optionsCopy.Remove(pair.Key); // Make sure we don't validate on invalid values
+                            continue; // Check if there are more invalid values, so that we can show those errors too
+                        }
```

Comment:
This could maybe be rewritten with if/else-if statements

```suggestion
                        }
                        else if (v < numericOption.Min || v > numericOption.Max)
                        {
                            errors.Add($"{kValidationErrorPrefix} '{pair.Key}' has out-of-bounds value '{o}' (must be between {numericOption.Min} and {numericOption.Max})");
                            optionsCopy.Remove(pair.Key); // Make sure we don't validate on invalid values
                        }
```

Reply:
I prefer the `continue` version, partially for the same reason as in the other `continue` comment. The other reason is that for validation, I prefer guard clauses over `else if` chains to prevent excessive code nesting, and to signal that invalid values requires the code to abort.